src/transformers/models/auto/image_processing_auto.py

```diff
@@ -430,10 +430,11 @@ def from_pretrained(cls, pretrained_model_name_or_path, *inputs, **kwargs):
 
         if image_processor_class is not None:
             # Update class name to reflect the use_fast option. If class is not found, None is returned.
-            if use_fast and not image_processor_class.endswith("Fast"):
-                image_processor_class += "Fast"
-            elif not use_fast and image_processor_class.endswith("Fast"):
-                image_processor_class = image_processor_class[:-4]
+            if use_fast is not None:
```

Comment:
We only update the image_processor_class if `use_fast` is explicitly set. This makes sure that if a fast image processor is saved out, it will be loaded in as a fast image processor from `AutoImageProcessor` by default

Reply:
I'm not sure I completely understood your comment, but let me try and explain the desired behaviour, and you can let me know if I've mistaken something :)

At the moment, we don't want `use_fast` to default to either `False` or `True`, as this will affect the image processor loaded in regardless of the type of image processor saved out. 

Desired behaviours
* Saved slow image processor -> loads slow image processor by default (`use_fast` unset)
* Saved fast image processor -> loads fast image processor by default (`use_fast` unset)
* Save slow image processor, `use_fast=False` -> loads slow image processor
* Save slow image processor, `use_fast=True` -> loads fast image processor
* Save fast image processor, `use_fast=False` -> loads slow image processor
* Save fast image processor, `use_fast=True` -> loads fast image processor

i.e. if a fast image processor is available, but a slow image processor is saved out, we still want to default to the slow image processor for now as they won't produce exactly the same outputs, and instead inform them that a fast version is available. This is to avoid unexpected behaviour

Reply:
OK, I got it now.

> that if a fast image processor is saved out, it will be loaded in as a fast image processor

which is not the current `main` but addressed in this PR.!

sorry (not easy to reason about all these cases and motivation)

Reply:
> which is not the current main but addressed in this PR.!

Yep! It was a bug that I hadn't noticed before 😬